msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/msalonly/basic/TestCase2517381.java

```diff
@@ -0,0 +1,163 @@
+
+        final MicrosoftStsPromptHandlerParameters promptHandlerParametersMsal = MicrosoftStsPromptHandlerParameters.builder()
+                .prompt(PromptParameter.SELECT_ACCOUNT)
+                .loginHint(username)
+                .sessionExpected(false)
+                .broker(mBroker)
+                .expectingBrokerAccountChooserActivity(false)
+                .expectingProvidedAccountInBroker(false)
+                .expectingLoginPageAccountPicker(false)
+                .expectingProvidedAccountInCookie(false)
+                .consentPageExpected(false)
+                .passwordPageExpected(true)
+                .speedBumpExpected(false)
+                .registerPageExpected(false)
+                .enrollPageExpected(false)
+                .staySignedInPageExpected(false)
+                .verifyYourIdentityPageExpected(false)
+                .howWouldYouLikeToSignInExpected(false)
+                .build();
+
+        String token = msalTestApp.acquireToken(username, password, promptHandlerParametersMsal,true);
+        Assert.assertNotNull(token);
+
+        // then acquire token silently and validate the token
+        msalTestApp.handleBackButton();
+        String silentToken = msalTestApp.acquireTokenSilent();
+        Assert.assertNotNull(silentToken);
+
+        // install old OneAuthTestApp then acquires token interactively and silently
+        final OneAuthTestApp oneAuthApp = new OneAuthTestApp();
+        oneAuthApp.installOldApk();
+        oneAuthApp.launch();
+        oneAuthApp.handleFirstRun();
+
+        final FirstPartyAppPromptHandlerParameters promptHandlerParametersOneAuth = FirstPartyAppPromptHandlerParameters.builder()
+                .broker(mBroker)
+                .prompt(PromptParameter.LOGIN)
+                .loginHint(username)
+                .consentPageExpected(false)
+                .speedBumpExpected(false)
+                .sessionExpected(false)
+                .expectingBrokerAccountChooserActivity(false)
+                .expectingLoginPageAccountPicker(false)
+                .enrollPageExpected(false)
+                .build();
+        oneAuthApp.addFirstAccount(username, password, promptHandlerParametersOneAuth);
+        oneAuthApp.confirmAccount(username);
+
+        // Hit back button to go on launch screen
+        oneAuthApp.handleBackButton();
+
+        final String silentTokenOneAuth = oneAuthApp.acquireTokenSilent();
+        Assert.assertFalse(TextUtils.isEmpty(silentTokenOneAuth));
+        oneAuthApp.assertSuccess();
+
+        // update msal test app
+        msalTestApp.update();
+        msalTestApp.launch();
+        msalTestApp.handleFirstRun();
+
+        // acquire token interactively and silently without prompting for creds
+        final String tokenAfterUpdatedMsal = msalTestApp.acquireToken(username, password, promptHandlerParametersMsal, false);
+        Assert.assertNotNull(tokenAfterUpdatedMsal);
+
+        msalTestApp.handleBackButton();
```

Comment:
These steps should not be required. The idea here is that user should acquireToken interactively and silently on the old apps first. Then after updating also, user should be able to do a silent call without having to do an interactive request.

Reply:
Yes, in this acquireToken step user only input username, after that the token will be successfully retrieved, no password needed. The forth parameter is to indicate whether the password page should show up.